Userland/Libraries/LibJS/Runtime/Intl/DisplayNames.cpp

```diff
@@ -150,8 +150,12 @@ ThrowCompletionOr<Value> canonical_code_for_display_names(GlobalObject& global_o
         if (!Unicode::is_type_identifier(code))
             return vm.throw_completion<RangeError>(global_object, ErrorType::OptionIsNotValidValue, code, "calendar"sv);
 
-        // b. Let code be the result of mapping code to lower case as described in 6.1.
-        // c. Return code.
+        // b. If code uses any of the backwards compatibility syntax described in Unicode Technical Standard #35 LDML § 3.3 BCP 47 Conformance, throw a RangeError exception.
+        if (code.contains('_'))
+            return vm.throw_completion<RangeError>(global_object, ErrorType::OptionIsNotValidValue, code, "calendar"sv);
```

Comment:
Does this also need to handle

- The subtag "root" to indicate the generic locale used as the parent of all languages in the CLDR data model ("und" can be used instead)
- The language tag may begin with a script subtag rather than a language subtag. This is specialized use only, and not required for CLDR conformance.
?

Reply:
I don't think so, not 100% sure, but my reasoning was:

* For the `root` restriction, that would apply to language tags, but the value here is only an extension keyword. It is valid to use `root` as an extension keyword within a locale, it just can't be the locale itself. i.e.:
```js
> new Intl.Locale('root')
Uncaught exception: [RangeError] root is not a structurally valid language tag
 -> Locale
 -> <unknown>
 -> (global execution context)
> new Intl.Locale('en-u-ca-root')
[Intl.Locale]
  locale: "en-u-ca-root"
  calendar: "root"
  numeric: false
```

Other engines (which already disallow underscores) also allow `root` here, it just doesn't get translated to anything. (I know we don't go by what other engines do, but it helped to double check my thinking here)

* Not beginning with a script subtag can't apply here because the type identifier is a superset of script subtags.